pkgs/top-level/cuda-packages.nix

```diff
@@ -0,0 +1,137 @@
+{ lib
+, pkgs
+, stdenv
+, cudaVersion
+}:
+
+with lib;
+
+(makeScope pkgs.newScope (self: let
```

Comment:
Hm, I'm not familiar with `makeScope` and `pkgs.newScope` yet.
Could you just briefly introduce the context: why is this needed and what are the costs?

Reply:
This is the standard way to create composible package sets and is used by all major package sets in Nixpkgs. At the bottom, you see `overrideScope'`. As a user, you can also easily override packages in the set recursively using this method.